obal/data/modules/check_copr_build.py

```diff
@@ -0,0 +1,54 @@
+"""
+Check if a build already exists in Copr
+"""
+
+import json
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.copr import copr_cli, CoprCliCommandError, full_name # pylint:disable=import-error,no-name-in-module
+
+
+def main():
+    """
+    Check if a build already exists in Copr
+    """
+    module = AnsibleModule(
+        argument_spec=dict(
+            user=dict(type='str', required=True),
+            project=dict(type='str', required=True),
+            nvr=dict(type='str', required=True),
+            package=dict(type='str', required=True),
+            config_file=dict(type='str', required=False),
+        )
+    )
+
+    user = module.params['user']
+    project = module.params['project']
+    nvr = module.params['nvr']
+    package = module.params['package']
+    config_file = module.params['config_file']
+
+    command = [
+        'get-package',
+        full_name(user, project),
+        '--name',
+        package,
+        '--with-all-builds'
+    ]
+
+    try:
+        package_info = json.loads(copr_cli(command, config_file=config_file))
+    except CoprCliCommandError as error:
+        if "Error: No package with name {} in copr {}".format(package, project):
+            module.exit_json(changed=True, exists=False)
+        else:
+            module.fail_json(msg='Retrieval of package from Copr failed', command=command, output=error.message,
+                             repo_name=full_name(user, project), package=package)
+
+    builds = (build for build in package_info['builds'] if build['state'] == 'succeeded')
+    exists = any("{}-{}".format(package, build['source_package']['version']) == nvr for build in builds)
```

Comment:
This is a lot to unpack, but I think this is equal to
```suggestion
    builds = (build for build in package_info['builds'] if build['state'] == 'succeeded')
    exists = any(builds)
```

Because a string is always true. So `package-False` will be `True`.

You could write it as
```suggestion
    exists = nvr in (build['source_package']['version'] for build in package_info['builds'] if build['state'] == 'succeeded')
```

Reply:
I am struggling to see how either of these two is equivalent to what I have. We need to compare whether the supplied nvr (name-version-release) is present in the set of builds that were returned for those builds marked as succeeded.

If you think this needs more breaking up of the logic, I could do this:

```
    successful_builds = (build for build in package_info['builds'] if build['state'] == 'succeeded')
    successful_nvrs = ("{}-{}".format(package, build['source_package']['version']) for build in builds)
    exists = nvr in successful_nvrs
```

Reply:
Ah, now I see it and I was misreading it. Your logic is sound, but I do think your proposed splitting makes sense. That's much easier to read.

Also brings me to another item: what about epochs? nvr doesn't include the epoch, so should we compare nevr?

Reply:
Epoch is a good question, I'm so used to Koji not reporting it. Looks like Copr does:

```
            "source_package": {
                "name": "rubygem-sinatra",
                "url": "https://download.copr.fedorainfracloud.org/results/@theforeman/foreman-nightly-scratch-789f37ac-5f59-5dc7-9412-b1357210f73f/srpm-builds/06166363/rubygem-sinatra-2.2.4-1.src.rpm",
                "version": "1:2.2.4-1"
            },
```

I will update the code to take Epoch into consideration.